### PR TITLE
Add Stages policy

### PR DESCRIPTION
**Pull request by cwilso**
(This includes changes from previous PR https://github.com/whatwg/sg/pull/222.)

### CODE REVIEW

**Review by tantek — APPROVED** [review]
LGTM

File: Stages.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+* <dfn>Stage</dfn>: each successive stage is closer to a landed spec and interoperable implementation.  These stages are tracked on the issues in the relevant WHATWG GitHub repository, using labels ("stage 0", "stage 1", etc.)
+* <dfn>Browser engine</dfn>: an independent implementation of the web platform (e.g. Chromium, WebKit and Gecko at present).
+* <dfn>Implementation interest</dfn>: a direct signal from a browser engine that a feature is on the roadmap to implement, as defined in [working mode](https://whatwg.org/working-mode#additions)].
+* <dfn>Browser engine representatives</dfn>: persons delegated by a browser engine to indicate implementation interest.
+* <dfn>Proposal champion</dfn>: person/organization that proposed and will advance a feature through the stages, typically including writing the specification, researching use cases, and prototyping.
+
+## Process
+
+* Any proposal effectively starts at Stage 0 without any approvals, by a community member filing a new issue on a relevant WHATWG spec.
```
Comment (annevk) [inline Stages.md:25]: Suggested addition:

* Anyone can review and submit feedback on Contributions.

File: Stages.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+* In order to drive a proposal forward, it is expected that a community member will step up to champion the proposal, through answering questions asynchronously, providing feedback when requested, and consideration of important open questions at triage meetings.
+* Stage level is tracked with labels on issues in the relevant specifications — “stage 0”, “stage 1”, etc.
+* Stage labels (other than stage 0) should only be added by browser engine representatives, or by Editors on the relevant specification, after showing the necessary support for a level.
```
Comment (annevk) [inline Stages.md:28]: Suggested addition:

* This process is subject to and governed by [WHATWG Policies](https://whatwg.org/policies), including the Intellectual Property Rights Policy.

**Review by annevk — COMMENTED** [review]
Thanks again for working on this. I think this is a valuable addition to the WHATWG.

The inline comments note a couple instances, but I would suggest we generally replace "feature" / "proposal" / ... with "Contribution" and link that to https://whatwg.org/ipr-policy#21-contribution. Similarly linking "Contributor" to https://whatwg.org/ipr-policy#contributor would be good (and consistently calling the person making the Contribution a Contributor).

That more clearly establishes how this works within our existing set of policies.

File: Stages.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
+# Stages
+
+The WHATWG's "document reality" approach has been effective for nailing down [fundamental parts of the platform](https://spec.whatwg.org/) and improving interoperability and developer satisfaction. However, our process can sometimes be daunting for newcomers, who don't know how to reliably get implementer feedback or editor time commitment. This can discourage contributions, and it can lead to frustration when implementer/editor feedback arrives late in the process after the contributor feels "close to done".
+
+The stage process described here is an additional policy tool designed to help provide a clear on-ramp for incubation efforts for contributors who aren't already editors, especially for additions to existing specs.  Stages are an optional, opt-in process that a feature proposer can use if they want to get more formal signals about how their feature is progressing. It asks for explicit implementer involvement at multiple stages, starting from notification that the problem is being worked on, then sign-off on the rough API and specification, and finally agreement on the full specification text.
```
[the remark this answers is not in the document]
Comment (cwilso) [reply Stages.md:5]: Addressed in https://github.com/whatwg/sg/pull/227/commits/5e55cc9a4bf501b6e4ac70c40e9376597316cc22.

File: Stages.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+The stage process described here is an additional policy tool designed to help provide a clear on-ramp for incubation efforts for contributors who aren't already editors, especially for additions to existing specs.  Stages are an optional, opt-in process that a feature proposer can use if they want to get more formal signals about how their feature is progressing. It asks for explicit implementer involvement at multiple stages, starting from notification that the problem is being worked on, then sign-off on the rough API and specification, and finally agreement on the full specification text. 
+
+Stage signals are also useful to the broader community, helping developers monitor the proposals that are moving through the various stages.  By explicitly signaling a proposal's progress, including implementer involvement, the community has a better idea of what is going on in the WHATWG.
+
+This tool is generally used for medium-to-large new features; it's not expected that explicit stage progression will be used for every feature.
```
[the remark this answers is not in the document]
Comment (cwilso) [reply Stages.md]: Addressed in https://github.com/whatwg/sg/pull/227/commits/bf6aedf26fd6034bbeb76e10e61d0844daaae903.

File: Stages.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
+# Stages
+
+The WHATWG's "document reality" approach has been effective for nailing down [fundamental parts of the platform](https://spec.whatwg.org/) and improving interoperability and developer satisfaction. However, our process can sometimes be daunting for newcomers, who don't know how to reliably get implementer feedback or editor time commitment. This can discourage contributions, and it can lead to frustration when implementer/editor feedback arrives late in the process after the contributor feels "close to done".
```
[the remark this answers is not in the document]
Comment (cwilso) [reply Stages.md]: Addressed in https://github.com/whatwg/sg/pull/227/commits/5e55cc9a4bf501b6e4ac70c40e9376597316cc22.

File: Stages.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+The stage process described here is an additional policy tool designed to help provide a clear on-ramp for incubation efforts for contributors who aren't already editors, especially for additions to existing specs.  Stages are an optional, opt-in process that a feature proposer can use if they want to get more formal signals about how their feature is progressing. It asks for explicit implementer involvement at multiple stages, starting from notification that the problem is being worked on, then sign-off on the rough API and specification, and finally agreement on the full specification text. 
+
+Stage signals are also useful to the broader community, helping developers monitor the proposals that are moving through the various stages.  By explicitly signaling a proposal's progress, including implementer involvement, the community has a better idea of what is going on in the WHATWG.
```
[the remark this answers is not in the document]
Comment (cwilso) [reply Stages.md]: Addressed in https://github.com/whatwg/sg/pull/227/commits/90f0c3247729ffcde00d14668750ddae1fe9d287.

File: Stages.md (hunk 1 above)
Comment (cwilso) [reply Stages.md:25]: Addressed in https://github.com/whatwg/sg/pull/227/commits/b8175528d6da72c791fb996a228ad13fde1f25e2.

File: Stages.md (hunk 2 above)
Comment (cwilso) [reply Stages.md:29]: Addressed in https://github.com/whatwg/sg/pull/227/commits/b8175528d6da72c791fb996a228ad13fde1f25e2.

**Review by annevk — COMMENTED** [review]
1. Cross-document links should be done as we do in other Markdown documents. The idea is that you can click around the Markdown documents and essentially stay in that "unpublished universe".
2. Terms like Contribution and Contributor need to be linked, as they are in other Markdown documents.
3. Please search for ".  " (two spaces) and replace with ". " (single space).
4. There's still quite a few occurrences of "feature" or "proposal" that should be "Contribution". And similar for "Contributor"-like terms. Generally the less unique terms we use the better.

If you want I'd be happy to do a pass and then you can leave feedback for me. Just say the word.

File: Stages.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
+# Stages
+
+The WHATWG's approach to "documenting reality" is ideal for nailing down [fundamental parts of the platform](https://spec.whatwg.org/) and improving interoperability and developer satisfaction. It can sometimes be daunting for newcomers, who don't know how to reliably get implementer feedback or editor time commitment. The WHATWG Stages process aims to help new and established Contributors get more streamlined feedback on their Contributions.
+
+The stage process described here is an additional policy tool designed to help provide a clear on-ramp for incubation efforts for contributors who aren't already editors, especially for additions to existing specs.  Stages are an optional, opt-in process that a feature proposer can use if they want to get more formal signals about how their feature is progressing. It asks for explicit implementer involvement at multiple stages, starting from notification that the problem is being worked on, then sign-off on the rough API and specification, and finally agreement on the full specification text.
```
[the remark this answers is not in the document]
Comment (cwilso) [reply Stages.md]: Tweaked to tighten language in https://github.com/whatwg/sg/pull/227/commits/3f1c9bc8f5306e9f3ceab0bde1778ce1c064be34, as part of the sentence was redundant but part wasn't.

File: Stages.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+
+Stage signals are also useful to the broader community, helping web developers monitor the Contribuions that are moving through the various stages.  By explicitly signaling a Contribution's progress, including implementer involvement, the community has a better idea of what is going on in the WHATWG.
+
+This tool is generally used for medium-to-large Contributions; it's not expected that explicit stage progression will be used for every Contribution.
+
+# Details
+
+This document formalizes some signal checkpoints for new features in existing WHATWG Living Standards. These checkpoints are modeled loosely on the [TC39 process](https://tc39.es/process-document/), which uses the concept of _stages_. Each subsequent stage implies a larger degree of consensus from the community, willingness to engage and/or implement and ship the feature in browser engines, and signals progress to web developers.  These checkpoints are not required for all features; this is simply a way to ensure that attention is appropriately drawn to incubations that have grown community support behind them.
```
[the remark this answers is not in the document]
Comment (cwilso) [reply Stages.md]: Collapsed in https://github.com/whatwg/sg/pull/227/commits/1cb92441c1bc00dd1e8269e3206c7e6e4f8926f2.  We don't need a separate section, and the intro was redundant, but the comparison to TC39 is still valid.

File: Stages.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+## Terminology
+
+* <dfn>Stage</dfn>: each successive stage is closer to a landed spec and interoperable implementation.  These stages are tracked on the issues in the relevant WHATWG GitHub repository, using labels ("stage 0", "stage 1", etc.)
+* <dfn>Browser engine</dfn>: an independent implementation of the web platform (e.g. Chromium, WebKit and Gecko at present).
+* <dfn>Implementation interest</dfn>: a direct signal from a browser engine that a feature is on the roadmap to implement, as defined in [working mode](https://whatwg.org/working-mode#additions)].
+* <dfn>Browser engine representatives</dfn>: persons delegated by a browser engine to indicate implementation interest.
+* <dfn>Proposal champion</dfn>: person/organization that proposed and will advance a feature through the stages, typically including writing the specification, researching use cases, and prototyping.
```
[the remark this answers is not in the document]
Comment (cwilso) [reply Stages.md]: Removed this term altogether.

**Comment by cwilso** [comment]
> * Cross-document links should be done as we do in other Markdown documents. The idea is that you can click around the Markdown documents and essentially stay in that "unpublished universe".

Done in https://github.com/whatwg/sg/pull/227/commits/97f07c14fe808db86ffda2a22fa163dc6cf78bfb.

> * Terms like Contribution and Contributor need to be linked, as they are in other Markdown documents.

Done in https://github.com/whatwg/sg/pull/227/commits/8d24ac275f61c3a4cedc2ce45e1125a5a2250e73, https://github.com/whatwg/sg/pull/227/commits/b2b46b7bcc2c36f05d80de89ce0be860472b4351, https://github.com/whatwg/sg/pull/227/commits/580c0838a3df5c73116dcd8bce566b599efc643d, https://github.com/whatwg/sg/pull/227/commits/7dac6951f60543ec0d74bbfcc9c66929d2cc36a9.

> * Please search for ".  " (two spaces) and replace with ". " (single space).

Done in https://github.com/whatwg/sg/pull/227/commits/96f035231d9984138693c84b361b951b133ca4e9.

> * There's still quite a few occurrences of "feature" or "proposal" that should be "Contribution". And similar for "Contributor"-like terms. Generally the less unique terms we use the better.

Fixed in https://github.com/whatwg/sg/pull/227/commits/a525b11076293334bc9a96614db2adfc310b814b, https://github.com/whatwg/sg/pull/227/commits/38523d048c8a61e0b7f4b10c3b6ff5218ee0f5fb, and https://github.com/whatwg/sg/pull/227/commits/37ffba494b35898c948a2839ddd195124c523b98.

Note there are still some intentional occurrences of "feature" and "proposal", when they are not specifically referring to the Contribution.

**Comment by cwilso** [comment]
Anne, you are more than welcome to take a pass at this directly if you would prefer.  Given your comment at last week's WHATNOT meeting, I'm going to go ahead and merge this soon, and we can further iterate if necessary.

**Comment by annevk** [comment]
@cwilso I'd like to be the one merging this unless you take responsibility for rolling it out on whatwg.org. I ran `convert_policy.py` myself and ran into an error, so I suspect you haven't actually tried that. (Thankfully the table seems to work fine.)

Please take a look at the changes I pushed.

**Comment by annevk** [comment]
I think we might need some kind of styling for the table actually. https://html5.org/temp/whatwg-stages.html looks pretty awful.

Minimally something like
```
table { border-collapse: collapse }
table td { vertical-align: top; }
table tr:nth-child(even), table thead tr { background: rgb(250, 250, 250); }
```
@domenic @zcorpan any thoughts on that?

**Comment by zcorpan** [comment]
Agreed more styling is needed, and I think some borders and padding aids readability further. On a narrow screen, the table overflows to the right, so setting a white background color is good.

```
table { border-collapse: collapse; background-color: white; }
table td { vertical-align: top; }
table td, table th { border: 1px solid #aaa; padding: 0.5em; }
table tr:nth-child(even), table thead tr { background: rgb(250, 250, 250); }
table td > :first-child { margin-top: 0; }
table td > :last-child { margin-bottom: 0; }
```

**Review by diekus — APPROVED** [review]
👍🏼

**Review by cwilso — COMMENTED** [review]
LGTM

**Review by annevk — APPROVED** [review]

**Pull request merged**